### PR TITLE
Added list comprehensions and stdlib tests

### PR DESCRIPTION
**Pull request by radrow**
This waits for https://github.com/aeternity/aesophia/pull/105 to be merged. 

It contains one commented out stdlib test with foldr – it seems to break FATE runtime

### CODE REVIEW

**Comment by radrow** [comment]
Some tests may be failing because the respective aesophia was not pinned yet in rebar.conf

File: test/contracts/stdlib_tests.aes [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    , (Func.tuplify2((a, b) => a + b)((2, 3)) == 5, "tuplify")
+
+    , (List.is_empty([]), "empty")
+    , (option_int_cmp(List.first([1,2,3,4]), Some(1)), "list first")
+    , (option_int_cmp(List.find((x) => x > 0, [-1, -2, 3, -1]), Some(3)), "find")
+    , (list_int_cmp(List.replace_at(2, 0, [1,1,1,1,1]), [1,1,0,1,1]), "replace at")
+    // following test seems to break the FATE runtime. It definitely SHOULD work.
+    // , (list_int_cmp(List.foldr((e, cont) => (x) => cont(e::x), (x) => x, [1,2,3])([]), [3,2,1]), "foldr")
+    , (List.foldl((a, x) => a - x, 0, [1,2,3]) == -6, "foldl")
+    , (list_int_cmp(List.reverse([1,2,3]), [3,2,1]), "erslist_int_cmp(e")
```
Comment (ThomasArts) [inline test/contracts/stdlib_tests.aes]: typo?

**Comment by lucafavatella** [comment]
Not planned for scope of 4.2.0

File: test/contracts/stdlib_tests.aes (hunk 1 above)
Comment (radrow) [reply test/contracts/stdlib_tests.aes]: Yep, it looks like I overused emacs macros

**Pull request merged**